tests/test_node.py

```diff
@@ -71,3 +72,46 @@ def test_run_as_root(
                     if record.levelno == logging.DEBUG:
                         found_expected_error = True
             assert found_expected_error
+
+    # An arbitrary time slice that is longer than it takes to spin up a cluster
+    # but avoids bash infinite loops.
+    @pytest.mark.timeout(60 * 15)
+    def test_popen_as_root(
+        self,
+        cluster_backend: ClusterBackend,
+        oss_artifact: Path,
+    ) -> None:
+        """
+        It is possible to run commands as root asynchronously
+        """
+        with Cluster(
+            agents=0,
+            public_agents=0,
+            cluster_backend=cluster_backend,
+            generate_config_path=oss_artifact,
+        ) as cluster:
+            (master, ) = cluster.masters
+            popen_1 = master.popen_as_root(
+                args=[
+                    '(mkfifo', '/tmp/pipe', '|', 'true)'
```

Comment:
I don't really understand this and I also am not sure why the original test we paired on does not work.

The test we wrote:

`popen_1: Loop until FOO exists; cat FOO`
`popen_2: Create FOO with content BAR`

Assert that `popen_1` and `popen_2` finish and `popen_1` includes the output `BAR`.

I'm happy to discuss this on Slack or voice chat if it is simpler.